devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -1276,27 +1292,30 @@ void HumanStateProvider::impl::selectJointsAndLinksForSecondaryCalibration(const
     }
 }
 
-void HumanStateProvider::impl::computeSecondaryCalibrationRotations(const std::vector<iDynTree::JointIndex>& jointZeroIndices, const std::vector<iDynTree::LinkIndex>& linkToCalibrateIndices)
+void HumanStateProvider::impl::computeSecondaryCalibrationRotationsForChain(const std::vector<iDynTree::JointIndex>& jointZeroIndices, const iDynTree::Transform& baseTransform, const std::vector<iDynTree::LinkIndex>& linkToCalibrateIndices)
 {
-    // setting to zero all the selected joints
+    // initialize vectors
     iDynTree::VectorDynSize jointPos(jointConfigurationSolution);
+    iDynTree::VectorDynSize jointVel(jointConfigurationSolution);
```

Comment:
Initialize with variable corresponding to joint velocity solution, if available

Reply:
Done https://github.com/robotology/human-dynamics-estimation/pull/211/commits/e7bbccb8cef6e44580587fa63930c23e05b4bb51